### PR TITLE
dd-9-trip-details

### PR DESCRIPTION
**Pull request by YuliyaVoronovich**
# Task

[Task](https://divas801801.atlassian.net/jira/software/projects/DD/boards/2?selectedIssue=DD-9)

# Description

-Add trip page
-Add the possibility of booking a place

A summary of the change, anything else that will to review this PR

## Pull request type

Please check the type of change your PR introduces:

- [x] 🍕 Feature
- [ ] 🐛 Bugfix
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update
- [ ] 🤖 Build related changes
- [ ] 🔥 Performance Improvements
- [ ] Other (please describe):

# Demo (situationally)

image

# Has It Been Tested?

- [ ] Yes
- [x] No, and this is why: maybe later...
- [ ] Tests are not needed for this task
- [ ] I need help with writing tests

### CODE REVIEW

File: src/app/features/admin/components/carriage-preview/carriage-preview.component.scss [hunk 1]
```diff
@@ -1,5 +1,9 @@
 @import '@taiga-ui/core/styles/taiga-ui-local';
 
+:host.selected-wrapper .seat.selected {
+  background-color: rgb(243 168 88);
```
Comment (RykovaMariia) [inline src/app/features/admin/components/carriage-preview/carriage-preview.component.scss]: var(--tui-status-warning) ?

File: src/app/features/admin/components/carriage-preview/carriage-preview.component.scss [hunk 2]
```diff
@@ -69,6 +75,14 @@ section {
   }
 }
 
+.seat.clickable {
+  cursor: pointer;
+}
+
+.seat.reserved {
+  background-color: rgb(182 178 178);
```
Comment (RykovaMariia) [inline src/app/features/admin/components/carriage-preview/carriage-preview.component.scss]: var(--tui-chart-categorical-04) ?

File: src/app/features/admin/components/carriage-preview/carriage-preview.component.ts [hunk 3]
```diff
@@ -28,6 +45,18 @@ export class CarriagePreviewComponent {
...
+    this.reservedSeats.clear();
+    const bookSeats = this.bookSeats();
+    if (!bookSeats) return;
+
+    const seatsForCurrentCarriage = bookSeats
+      .filter((seat) => seat.carriageType === this.carriage()?.name)
+      .map((seat) => seat.localSeatNumber);
+
+    seatsForCurrentCarriage.forEach((seatNumber) => this.reservedSeats.add(seatNumber));
+  }
```
Comment (RykovaMariia) [inline src/app/features/admin/components/carriage-preview/carriage-preview.component.ts]:
```suggestion
      .forEach((seat) => this.reservedSeats.add(seat.localSeatNumber));
```

File: src/app/features/admin/components/carriage-preview/carriage-preview.component.ts [hunk 4]
```diff
@@ -54,4 +83,24 @@ export class CarriagePreviewComponent {
...
     );
   }
+
+  public onSeatClick(seatNumber: number) {
+    if (!this.options().isClick) return;
+    if (this.selectedSeat === seatNumber) {
+      this.selectedSeat = null;
+    } else {
+      this.selectedSeat = seatNumber;
+    }
```
Comment (RykovaMariia) [inline src/app/features/admin/components/carriage-preview/carriage-preview.component.ts]:
```suggestion
   this.selectedSeat = this.isSeatSelected(seatNumber) ? null : seatNumber;
```

File: src/app/features/search/pages/search-detail-page/search-detail-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+  public readonly isBookSeat = this._isBookSeat.asReadonly();
+
+  public orderId = signal(0);
+
+  public options = signal({ isClick: true, isShowTitle: false });
+
+  public fromStation!: number;
+
+  public toStation!: number;
```
Comment (ChickenMcNuggetss) [inline src/app/features/search/pages/search-detail-page/search-detail-page.component.ts]: Is it really okay to use non-null assertion operator here? Is it possible that fromStation or toStation could be null??

File: src/app/features/search/components/order-panel/order-panel.component.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      },
+      { allowSignalWrites: true }
+    );
+  }
+
+  public close() {
+    this.open.set(false);
+  }
+
+  public hendlerBookSeat() {
```
Comment (ChickenMcNuggetss) [inline src/app/features/search/components/order-panel/order-panel.component.ts]: handlerBookSeat

File: src/app/features/search/pages/search-detail-page/search-detail-page.component.ts (hunk 5 above)
Comment (ChickenMcNuggetss) [reply src/app/features/search/pages/search-detail-page/search-detail-page.component.ts]: Okay I see the point now

File: src/app/features/search/pages/search-detail-page/search-detail-page.component.ts [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+
+  public carriages = this.carriageService.carriages;
+
+  private readonly _ride = signal<Trip | null>(null);
+
+  public readonly ride = this._ride.asReadonly();
+
+  private readonly _isBookSeat = signal(false);
+
+  public readonly isBookSeat = this._isBookSeat.asReadonly();
```
Comment (RykovaMariia) [inline src/app/features/search/pages/search-detail-page/search-detail-page.component.ts]: there is no need to do this inside the component

File: src/app/features/admin/components/carriage-preview/carriage-preview.component.ts [hunk 8]
```diff
@@ -1,25 +1,42 @@
...
   selector: 'dd-carriage-preview',
   standalone: true,
-  imports: [NgFor, TuiCardLarge, TuiSurface, TuiTitle, TuiHeader, TuiAppearance],
+  imports: [NgFor, TuiCardLarge, TuiSurface, TuiTitle, TuiHeader, TuiAppearance, NgClass],
   templateUrl: './carriage-preview.component.html',
   styleUrl: './carriage-preview.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CarriagePreviewComponent {
+  public options = input<{ isClick: boolean; isShowTitle: boolean }>({
```
Comment (ChickenMcNuggetss) [inline src/app/features/admin/components/carriage-preview/carriage-preview.component.ts:17]: { isClick: boolean; isShowTitle: boolean } - seems like interface

File: src/app/features/search/components/order-panel/order-panel.component.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+  public isBookSeat = input.required<boolean>();
+
+  public bookSeat = output<SelectedOrder>();
+
+  private readonly authService = inject(AuthService);
+
+  private readonly alert = inject(AlertService);
+
+  public isLoggedIn = this.authService.isLoggedIn;
```
Comment (Dewlar) [inline src/app/features/search/components/order-panel/order-panel.component.ts]: private readonly ?

Comment (Dewlar) [reply src/app/features/search/components/order-panel/order-panel.component.ts]: you use it once
maybe you shouldn't create another variable?
but use the service? `this.authService.isLoggedIn()`
and delete this variable

File: src/app/features/search/components/order-panel/order-panel.component.ts [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+      { allowSignalWrites: true }
+    );
+  }
+
+  public close() {
+    this.open.set(false);
+  }
+
+  public handlerBookSeat() {
+    if (this.isLoggedIn()) {
```
Comment (Dewlar) [inline src/app/features/search/components/order-panel/order-panel.component.ts]: `this.authService.isLoggedIn()` ?
```suggestion
    if (this.authService.isLoggedIn()) {
```

File: src/app/features/search/components/order-panel/order-panel.component.ts (hunk 9 above)
Comment (YuliyaVoronovich) [reply src/app/features/search/components/order-panel/order-panel.component.ts:27]: It’s like this everywhere...why not?

File: src/app/features/search/pages/search-detail-page/search-detail-page.component.html [hunk 11]
```diff
@@ -0,0 +1,53 @@
+<dd-order-panel
+  [selectedOrder]="selectedOrder"
+  [isBookSeat]="isBookSeat()"
+  (bookSeat)="bookSeat($event)"
+></dd-order-panel>
+
+<div tuiHeader class="header">
```
Comment (Dewlar) [inline src/app/features/search/pages/search-detail-page/search-detail-page.component.html:7]: `tuiHeader` - not imported

**Review by Dewlar — DISMISSED** [review]

File: src/app/features/admin/components/carriage-preview/carriage-preview.component.ts (hunk 8 above)
Comment (YuliyaVoronovich) [reply src/app/features/admin/components/carriage-preview/carriage-preview.component.ts:17]: This is just an object so as not to make 2 separate input signals. these variables are not particularly related to each other. is there a need for an interface?

Comment (ChickenMcNuggetss) [reply src/app/features/admin/components/carriage-preview/carriage-preview.component.ts:17]: Ok, if it seems useless then leave it just like that👌🏻

**Review by ChickenMcNuggetss — APPROVED** [review]

**Review by RykovaMariia — APPROVED** [review]

**Pull request merged**